Metrics forwarder pko setup (#3)
openshift/hypershift-dataplane-metrics-forwarder: README.md

> # Metrics Forwarder

Reviewer: ```suggestion
# Dataplane Metrics Forwarder
```

> The Metrics Forwarder is a Proxy that lives in each Hosted Control Plane (HCP) Namespace in Management Clusters. The main goal of this Proxy is to let the relevant Hosted Cluster's Data Plane push metrics to RHOBS. This proxy exposes a route that is known to the Cluster Monitoring Operator (CMO) running in the Data Plane. The CMO pushes the metrics to this route. And the proxy simply forwards this request to the Openshift Observability Operator (OBO) running on the relevant Management Cluster. This way the metrics ultimately end up in RHOBS.

Reviewer: ```suggestion
The dataplane metrics forwarder is a proxy that lives in each hosted control plane (HCP) namespace in management clusters. The main goal of this proxy is to let the hosted clusters push metrics to RHOBS. This proxy exposes a route that is known to the Cluster Monitoring Operator (CMO) running in the dataplane. The CMO pushes the metrics to this route. And the proxy simply forwards this request to the OpenShift Observability Operator (OBO) running on the relevant management cluster. This way the dataplane metrics ultimately end up in RHOBS.
```

## Deployment process

> The proxy is deployed using ACM policy. The ACM Policy targets all Management Clusters and only HCP Namespaces in those Management Clusters. It deploys a PKO Package which deploys the resources the proxy needs. The resources it deploys are:

Reviewer: ```suggestion
The proxy is deployed using ACM policy. The ACM policy targets all management clusters and only HCP namespaces in those management clusters. It deploys a PKO package which deploys the resources the proxy needs. The resources it deploys are:
```

- ServiceAccount
- Role
- RoleBinding
- CronJob
- Issuer
- Certificate
- ConfigMap
- Deployment
- Service
- Route

> Additionally, the CMO on the Data Plane is configured using another Policy that only targets Hosted Clusters.

Reviewer: ```suggestion
Additionally, the CMO on the dataplane is configured using another policy that only targets hosted clusters.
```

Reviewer 2: policy -> ACM policy?

## Proxy initialization process

> The Proxy needs to be initialized to work properly. This phase might not be needed later if the root-ca contains certificates that are compatible with Cert Manager.
> Initially, PKO Package deploys 4 resources:

Reviewer: ```suggestion
The proxy needs to be initialized to work properly. This phase might not be needed later if the root-ca contains certificates that are compatible with [cert-manager](https://cert-manager.io/).
Initially, PKO package deploys 4 resources:
```

- ServiceAccount
- Role
- RoleBinding
- CronJob

> This CronJob mianly recreates the root-ca in the HCP with names that are compatible with Cert Manager. 

Reviewer: ```suggestion
This CronJob mainly recreates the root-ca in the HCP with names that are compatible with cert-manager. 
```

> ## Contact

Reviewer 2: Let's remove this part? since the channel are redhat internal slack channel.

Reviewer 2: We can ignore that now.
